src/cmd/flux-mini.py

```diff
@@ -0,0 +1,305 @@
+#!/usr/bin/env python
+
+##############################################################
+# Copyright 2019 Lawrence Livermore National Security, LLC
+# (c.f. AUTHORS, NOTICE.LLNS, COPYING)
+#
+# This file is part of the Flux resource manager framework.
+# For details, see https://github.com/flux-framework.
+#
+# SPDX-License-Identifier: LGPL-3.0
+##############################################################
+
+from __future__ import print_function
+
+import re
+import os
+import sys
+import math
+import logging
+import argparse
+import json
+import collections
+import flux
+from flux import job
+from datetime import timedelta
+
+try:
+    collectionsAbc = collections.abc
+except AttributeError:
+    collectionsAbc = collections
+
+from flux import util
+
+
+class JobSpec:
+    def __init__(self, command, num_tasks=1, cores_per_task=1, num_nodes=None):
+        """
+        Constructor builds the minimum legal v1 jobspec.
+        Use setters to assign additional properties.
+        """
+        if type(command) is not list:
```

Comment:
I think this is close to what you want, but not exactly.  Python's duck typing means you'll trim out a bunch of things that act like lists but aren't explicitly lists (e.g., `tuple`, `deque`).   ([relevant stackoverflow answer](https://stackoverflow.com/a/37842328)).   In `_create_resource`, there is an example of checking for that it is `collectionsAbc.Sequence` and not a `str`.  (Although now that I look at that again, it only works for python 3 and not python 2....grr....maybe we should wrap that functionality up into a utility function.....feel free to not include that in this PR, we can always open an issue and loop back around to it).

Reply:
One other thing, you'll want to switch to using `isintance` rather than `type` to account for inheritance ([relevant stackoverflow post](https://stackoverflow.com/a/1549854)).

Reply:
OK, I'll use isinstance and try to be a bit more inclusive but hold off on the collectionsAbc stuff.

Reply:
@trws : any insight on the "right way" to test for a thing that maps to a `list` in YAML?

I think the "Sequence not str" check above also allows buffers and other unintended types in.

Reply:
I see this is now assertions etc.  While that’s fine, we could be using typing comments and mypy if we are doing type-checking.

As to actually telling if it will become a list, the only way I know to be sure is to actually turn the thing into yaml and back and check, since types can register their own serializers.  Just checking that it’s a sequence and not a string seems reasonable.